File: 프로그래머스/lv1/42862. 체육복/체육복.kt
class Solution {
    fun solution(n: Int, lost: IntArray, reserve: IntArray): Int {
        val lostStudent = lost.sorted().filter { reserve.contains(it).not() }.toMutableList()

        reserve.filter { lost.contains(it).not() }.sorted().forEach {
            if (lostStudent.contains(it - 1)) {
                lostStudent.remove(it - 1)
            } else if (lostStudent.contains(it + 1)) {
                lostStudent.remove(it + 1)
            }
        }

        return n - lostStudent.size
    }
}

File: 프로그래머스/lv1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    _lost = [l for l in lost if l not in reserve]
    _reserve = [r for r in reserve if r not in lost]
    _reserve.sort()
    
    for r in _reserve:
        if r - 1 in _lost:
            _lost.remove(r - 1)
        elif r + 1 in _lost:
            _lost.remove(r + 1)
                    
    return n - len(_lost)